File: app/src/main/kotlin/com/kondenko/pocketwaka/api/model/stats/StatsDataWrapper.java

package com.kondenko.pocketwaka.api.model.stats;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class StatsDataWrapper implements Parcelable {

    @SerializedName("data")
    @Expose
    public Stats stats = null;

    public final static Parcelable.Creator<StatsDataWrapper> CREATOR = new Creator<StatsDataWrapper>() {
        @SuppressWarnings({
                "unchecked"
        })
        public StatsDataWrapper createFromParcel(Parcel in) {
            StatsDataWrapper instance = new StatsDataWrapper();
            instance.stats = ((Stats) in.readValue((Stats.class.getClassLoader())));
            return instance;
        }

        public StatsDataWrapper[] newArray(int size) {
            return (new StatsDataWrapper[size]);
        }

    };

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(stats);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/kotlin/com/kondenko/pocketwaka/data/stats/model/StatsDataWrapper.kt
package com.kondenko.pocketwaka.data.stats.model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class StatsDataWrapper : Parcelable {

    @SerializedName("data")
    @Expose
    lateinit var stats: Stats

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(stats)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        val CREATOR: Parcelable.Creator<StatsDataWrapper> = object : Parcelable.Creator<StatsDataWrapper> {
            override fun createFromParcel(`in`: Parcel): StatsDataWrapper {
                val instance = StatsDataWrapper()
                instance.stats = `in`.readValue(Stats::class.java.classLoader) as Stats
                return instance
            }

            override fun newArray(size: Int): Array<StatsDataWrapper?> {
                return arrayOfNulls(size)
            }

        }
    }

}
